docs/learn/integrations/dagster.md

```diff
@@ -0,0 +1,587 @@
+
+```python
+from metaxy.ext.dagster import build_asset_in_from_feature
+
+
+@dg.asset(ins=build_asset_in_from_feature(UpstreamFeature, input_name="upstream"))
+def downstream_asset(upstream):
+    # upstream is an Increment with added/changed/removed
+    print(f"Processing {upstream.added.shape[0]} new samples")
+```
+
+## Parallelization Patterns
+
+Metaxy's Dagster integration supports three parallelization strategies:
+
+### 1. Data Parallel (Default)
+
+Process all documents/samples at once. This is the default mode.
+
+```python
+io_manager = MetaxyIOManager.from_store(store_resource)
+```
+
+**Use when:**
+
+- Processing small to medium datasets
+- Your compute resources can handle full dataset
+- You want simple, straightforward processing
+
+### 2. Event/Key Parallel
+
+Process **one entity end-to-end** through the entire feature graph. Perfect for processing individual videos, documents, or users through your pipeline.
+
+```python
+from metaxy.ext.dagster import (
+    MetaxyIOManager,
+    create_video_partitions_def,
+    filter_samples_by_partition,
+)
+
+# Setup once
+video_partitions = create_video_partitions_def(name="videos")
+
+io_manager = MetaxyIOManager(store=store_resource, partition_key_column="video_id")
+
+
+# Root feature: Manual partition filtering
+@dg.asset(partitions_def=video_partitions)
+def raw_video(context: dg.AssetExecutionContext, store):
+    video_id = context.partition_key  # Current video being processed
+
+    # Generate all samples, then filter to this video
+    all_samples = generate_samples()
+    samples = filter_samples_by_partition(
+        all_samples, partition_column="video_id", partition_key=video_id
+    )
+
+    increment = store.resolve_update(RawVideo, samples=samples)
+    store.write_metadata(RawVideo, increment.added)
+    return RawVideo.spec()
+
+
+# Downstream feature: Automatic filtering
+@dg.asset(partitions_def=video_partitions)
+def clean_video(context: dg.AssetExecutionContext, store):
+    video_id = context.partition_key
+
+    # IOManager automatically filters increment to this video!
+    increment = store.resolve_update(CleanVideo)
+    results = process(increment.added)
+    store.write_metadata(CleanVideo, results)
+    return CleanVideo.spec()
+```
+
+**Use when:**
+
+- Processing large datasets that don't fit in memory
+- Each entity can be processed independently
+- You want fine-grained retry and monitoring per entity
+- Processing videos, documents, or users one at a time
+
+**How it works:**
+
+1. **Root features**: Use `filter_samples_by_partition()` to filter samples to current partition
+2. **Downstream features**: IOManager automatically filters `Increment` to current partition
+3. **Dagster**: Orchestrates each partition as a separate run
+
+### 3. Subsampled
+
+Process a filtered subset of data. Useful for testing and development.
+
+```python
+# Limit to first N samples
+io_manager = MetaxyIOManager(store=store_resource, sample_limit=100)
```

Comment:
can we ensure that the samples are consistent i.e. then choosing 10 out of 100 documents that the same 10 key prefixes are then applied for all downstream computations? I.e. that we not consistely re-sample 10 random ones from the upstream primary metadata store